canonical-web-and-design/maas-ui
# Domain resource edit

## Done

- Adds a form to resources table to edit a resource
- Drive-by: only loads single domain in header instead of fetch to prevent refreshing domains list with empty resources

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- `yarn start`
- go to http://0.0.0.0:8400/MAAS/r/domain/2
- edit some resource records in the table
- try to pass data that is not IP address into record type A to see error from server
- form should work as expected when sumbitting, canceling, etc

## Fixes

Fixes: canonical-web-and-design/app-squad#51
